topcoffea/modules/corrections.py

```diff
@@ -21,24 +21,31 @@ def get_method1a_wgt_singlewp(eff,sf,passes_tag):
     wgt = p_data/p_mc
     return wgt
 
-# Evaluate btag sf from central correctionlib json
-def btag_sf_eval(jet_collection,wp,year,method,syst):
+def get_method1a_wgt_doublewp(effA, effB, sfA, sfB, cutA, cutB, cutC):
+    effA_data = effA * sfA
+    effB_data = effB * sfB
+
+    pMC = ak.prod(effA[cutA], axis=-1) * ak.prod(effB[cutB] - effA[cutB], axis=-1) * ak.prod(1 - effB[cutC], axis=-1)
+    pMC = ak.where(pMC==0,1,pMC) # removeing zeroes from denominator...
+    pData = ak.prod(effA_data[cutA], axis=-1) * ak.prod(effB_data[cutB] - effA_data[cutB], axis=-1) * ak.prod(1 - effB_data[cutC], axis=-1)
+
+    return pData, pMC
 
+# Evaluate btag sf from central correctionlib json
+def btag_sf_eval(jet_collection,wp,year,method,syst,fromPOG=True):
     # Get the right sf json for the given year
-    if year == "2016APV":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2016preVFP_UL_btagging.json")
-    elif year == "2016":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2016postVFP_UL_btagging.json")
-    elif year == "2017":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2017_UL_btagging.json")
-    elif year == "2018":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2018_UL_btagging.json")
-    elif year == "2022":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2022_btagging.json")
-    elif year == "2022EE":
-        fname = topcoffea_path("data/btag_sf_correctionlib/2022EE_btagging.json")
+    runII = ("2016", "2017", "2018")
+
+    clib_year = year
+    if year.startswith("2016"):
+        clib_year = "2016preVFP" if (year == "2016APV" or year == "2016preVFP") else "2016postVFP"
+
+    clib_year = clib_year + "_UL" if any(runIIyear in clib_year for runIIyear in runII) else clib_year
+
+    if fromPOG is True:
+        fname = topcoffea_path(f"data/POG/BTV/{clib_year}/btagging.json.gz")
```

Comment:
As of now, I think this will not properly grab the Run 3 files in the `data/POG/BTV/` directory. I would propose the following as a quick fix:

```
if year.startswith("2022"):
    clib_year = "2022_Summer22EE" if (year == "2022EE") else "2022_Summer22"
if year.startswith("2023"):
    clib_year = "2023_Summer23BPix" if (year == "2023BPix") else "2023_Summer23"
```

This will default the 2022 and 2022EE years to the correct directory. Also I think this would allow us to remove the other correctionlib files that were previously being used.

Reply:
Good idea, I have implemented a dict to collect the correspondence between years and tags for correction-lib, so we can avoid using multiple if statements. In addition, I adopted the same strategies for the PU SFs